tests/SearchPluginTest.php

```diff
@@ -491,4 +491,39 @@ public function testSearchQueryInvalidWhere() {
 		$this->assertEquals(400, $response->getStatus());
 	}
 
+	public function testSearchQueryInfiniteLoopEmptyLiteral() {
+		$this->searchBackend->expects($this->any())
+			->method('getArbiterPath')
+			->willReturn('foo');
+
+		$plugin = new SearchPlugin($this->searchBackend);
+		$server = new Server();
+		$plugin->initialize($server);
+
+		$request = new Request('SEARCH', '/index.php/foo', [
+			'Content-Type' => 'text/xml'
+		]);
+		$request->setBaseUrl('/index.php');
+		$request->setBody(fopen(__DIR__ . '/infiniteloopemptyliteral.xml', 'r'));
```

Comment:
Please add the xml with the example request

Reply:
Oh... I forgot to add the request data.